app/license.go

```diff
@@ -245,3 +257,65 @@ func (s *Server) RequestTrialLicense(trialRequest *model.TrialLicenseRequest) *m
 
 	return nil
 }
+
+// GenerateRenewalToken returns the current active token or generate a new one if
+// the current active one has expired
+func (s *Server) GenerateRenewalToken(expiration time.Duration) (string, *model.AppError) {
+	license := s.License()
```

Comment:
Should we restrict creating a renewal link if the license is cloud? Doesn't seem to make sense for that case

Reply:
I'm 1/5 on that. I'd add it to be consistent